### PR TITLE
update js.dart to match dartlang/js.dart

### PR DESCRIPTION
**Pull request by danrubel**
- update js.dart to match dartlang/js.dart
- move `EventListener` into `disposables.dart`

@devoncarew

### CODE REVIEW

File: lib/src/js.dart [hunk 1]
```diff
@@ -10,9 +11,12 @@ import '../utils/disposable.dart';
 
 export 'dart:js' show context, JsObject, JsFunction;
 
+final JsObject _browserWindow = new JsObject.fromBrowserObject(context['window']);
+final JsObject _browserJson = _browserWindow['JSON'];
+
 Logger _logger = new Logger("js");
 
-dynamic jsify(obj) {
+JsObject jsify(obj) {
```
Comment (devoncarew) [inline lib/src/js.dart]: I think this should be dynamic - to support things like int, bool and String.

File: lib/src/js.dart [hunk 2]
```diff
@@ -20,6 +24,32 @@ dynamic jsify(obj) {
   return obj;
 }
 
+JsObject uncrackDart2js(dynamic obj) => context.callMethod('uncrackDart2js', [obj]);
```
Comment (devoncarew) [inline lib/src/js.dart:28]: We should add a comment here that this method depends on the embedder defining this method in their JS (`uncrackDart2js`).

**Comment by devoncarew** [comment]
lgtm modulo what I've noted

File: lib/src/js.dart (hunk 1 above)
Comment (danrubel) [inline lib/src/js.dart]: Ah. Good point. Done.

File: lib/src/js.dart (hunk 2 above)
Comment (danrubel) [inline lib/src/js.dart:28]: Done.

**Comment by danrubel** [comment]
Merged

**Pull request closed without merging**